Add grpc checks support for esm.

DaryaShirokova opened the pull request:
## PCI review checklist

<!-- heimdall_github_prtemplate:grc-pci_dss-2024-01-05 -->

- [X] I have documented a clear reason for, and description of, the change I am making.

Adding grpc health check, which is already supported in consul, to consul-esm.

- [X] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.

Can be reverted.

- [X] If applicable, I've documented the impact of any changes to security controls.

NA.

----
Add gRPC health check support to Consul ESM:

 * Implement gRPC health check type alongside existing HTTP and TCP checks
 * Refactor check-stopping logic: extract shared stopExistingChecks method from duplicated inline code in updateCheckHTTP and updateCheckTCP
* Add tlsConfigsEqual helper for TLS config comparison
* Add gRPC test server utilities and tests for healthy, unhealthy, and TLS scenarios

DaryaShirokova committed:
Add grpc checks support for esm.

santoshpulluri commented on check_test.go:
```diff
@@ -298,6 +560,230 @@ func TestCheck_MinimumInterval(t *testing.T) {
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer s.Stop()
```
defer s.Stop() executes when `setUp := func(t *testing.T) *fixture` returns, so the Consul test server is gone before the subtest uses `runner: &CheckRunner`. That makes these cases validate behavior against a dead server rather than stopExistingChecks. Please switch this to `t.Cleanup(s.Stop)` or keep the server on the fixture and stop it at subtest cleanup time.

DaryaShirokova replied:
Should be resolved now, thanks for the comment!

santoshpulluri commented:
Please update the PR description with appropriate details.
README files needs to be updated to mention the gRPC support with this change.

The changes look good except for the minor comment related to test case.

DaryaShirokova committed:
Fix test server lifecycle in setUp and add gRPC check example to README

DaryaShirokova commented:
> Please update the PR description with appropriate details. README files needs to be updated to mention the gRPC support with this change.
> 
> The changes look good except for the minor comment related to test case.

Thanks @santoshpulluri , I have addressed the feedback, could you please have another look?

santoshpulluri reviewed (approved):
I reviewed the latest updates. The earlier test lifecycle issue has been addressed, and the README now includes a gRPC check example. The gRPC support is integrated cleanly into the existing check runner flow, reuses Consul’s native gRPC health check implementation, and I don’t see any remaining blocking concerns. Approved.

DaryaShirokova commented:
@santoshpulluri thanks for the review! are there any action items on my side before it could be merged or you'll merge it on your side?
